vhdbuilder/packer/post-install-dependencies.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+OS=$(sort -r /etc/*-release | gawk 'match($0, /^(ID_LIKE=(coreos)|ID=(.*))$/, a) { print toupper(a[2] a[3]); exit }')
+UBUNTU_OS_NAME="UBUNTU"
+
+source /home/packer/provision_installs.sh
+source /home/packer/provision_installs_distro.sh
+source /home/packer/provision_source.sh
+source /home/packer/provision_source_distro.sh
+source /home/packer/tool_installs.sh
+source /home/packer/tool_installs_distro.sh
+
+CPU_ARCH=$(getCPUArch)  #amd64 or arm64
+VHD_LOGS_FILEPATH=/opt/azure/vhd-install.complete
+
+if [[ $OS == $MARINER_OS_NAME ]]; then
+  if grep -q "kata" <<< "$FEATURE_FLAGS"; then
+    enableMarinerKata
+  else
+    # Leave automatic package update disabled for the kata image
+    enableDNFAutomatic
+  fi
+fi
+
+if [[ $OS == $UBUNTU_OS_NAME ]]; then
+  # shellcheck disable=SC2021
+  current_kernel="$(uname -r | cut -d- -f-2)"
+  dpkg --get-selections | grep -e "linux-\(headers\|modules\|image\)" | grep -v "$current_kernel" | tr -s '[[:space:]]' | tr '\t' ' ' | cut -d' ' -f1 | xargs -I{} apt-get remove -yq {}
+
+  # strip old kernels/packages
+  apt-get -y autoclean || exit 1
+  apt-get -y autoremove --purge || exit 1
+  apt-get -y clean || exit 1
+  # remove apport
+  apt-get purge --auto-remove apport open-vm-tools -y
```

Comment:
do this before autoclean/autoremove/clean so those commands catch the transitive deps which will no longer be required